autoextract/stats.py

```diff
@@ -49,7 +76,7 @@ def error_ratio(self):
 
     @zero_on_division_error
     def success_ratio(self):
-        return self.n_results / (self.n_results + self.n_fatal_errors)
+        return self.n_extracted_queries / self.n_input_queries
```

Comment:
Hey @ivanprado! I tried using the CLI with the updated code, and this changes the meaning of success in the progress bar. I wanted success to be defined as "% of input queries which we were able to get the results for" - basically, success rate as measured in the output file. It seems now this % is lower, as it includes unsuccessful retries. 

For example, if there are 100 input queries, each required 2 attempts, but it was possible to get everything, previously success was 100% (as we were able to get all results). Now, if I'm not mistaken, success is going to be 50% (as half of requests were retries).

On the other hand, it'd be nice to include query errors to success calculation. Previously they were not included.

So, proposal:

1. Define success as % of non-error results in the output file.
2. If that's important, measure "retry overhead" separately.

What do you think?

Reply:
Another example: user increases max-queery-retries. This is likely to increase % of extracted results, but also increase % of retries even more. So a lower success rate will be reported in the CLI, which may motivate user not to enable retries.

Reply:
> I wanted success to be defined as "% of input queries which we were able to get the results for"

Agree. This is what I wanted the code to do, but maybe there is a bug anywhere. 

> It seems now this % is lower, as it includes unsuccessful retries.

Why does it include unsuccessful retries? `n_input_queries` should be only counting input queries and skipping the retries logic here: https://github.com/scrapinghub/scrapinghub-autoextract/blob/master/autoextract/aio/client.py#L250 

Besides `n_extracted_queries` should only be counting successful responses, which are not being retried. So, I don't see retries affecting here as well. 

Regarding your proposal, point number (1) I think it is already achieved (unless there is a bug I'm not able to see). And point number (2) of "retry overhead" is something is also being computed. See the following example stats:

 ```
Summary
-------
Mean connection time:     0.35
Mean response time:       10.07
Throttle ratio:           0.0%
Attempts:                 11
Errors:                   90.9%, fatal: 0, non fatal: 10
Successful URLs:          2 of 11
Success ratio:            18.2%
Billable query responses: 18 of 19
```

That's the way of reading them: Sucess ratio is 18% because only 2 of the 11 URLs in the input file were successfully extracted (free of any kind of error). But it required sending 19 queries in 11 request attempts (batch size is 2 here). So the overhead in queries here is 19/11, and the overhead in requests is 11/6 (6 requests it the minimum number of requests possible if no errors to obtain 11 queries responses).

Something that can be confusing is that `n_input_queries` computing is ahead of `n_extracted_queries` so **the impression, if you don't wait until the end, is that success rate is way low**. But it will eventually catch up at the end. I didn't find a quick solution for that so I left as is, but maybe there is a way to improve it somehow. 

Finally, I think there is room for improvement in the the `tqdm` error section. I don't understand the distinction between `erros` and `fatal_errors`, and also it will be very confusing for the user. But I didn't change things there as I didn't get the full idea of the errors (and I full of work for this week, I decided not going deeper ;-) ).

Reply:
I fixed the problem of showing underestimated success rate in `tqdm` in this PR: https://github.com/scrapinghub/scrapinghub-autoextract/pull/25/files. It was just a matter of moving a line of code. Thank you @kmike for the inspiration.